tests/Avalonia.Markup.Xaml.UnitTests/Xaml/AssignBindingTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿#nullable enable
+
+using Avalonia.Controls;
+using Avalonia.Data;
+using Avalonia.UnitTests;
+using Xunit;
+
+namespace Avalonia.Markup.Xaml.UnitTests.Xaml;
+
+public class AssignBindingTests : XamlTestBase
+{
+    [Fact]
+    public void AssignBinding_Works_With_Clr_Property()
+    {
+        using var app = UnitTestApplication.Start(TestServices.StyledWindow);
+
+        var control = (AssignBindingTestControl)AvaloniaRuntimeXamlLoader.Load(
+            """
+            <local:AssignBindingTestControl
+                xmlns='https://github.com/avaloniaui'
+                xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'
+                xmlns:local='clr-namespace:Avalonia.Markup.Xaml.UnitTests.Xaml;assembly=Avalonia.Markup.Xaml.UnitTests'
+                ClrBinding='{Binding SomePath}' />
+            """);
+
+        Assert.NotNull(control.ClrBinding);
+    }
+
+    [Fact]
+    public void AssignBinding_Works_With_AttachedProperty()
+    {
+        using var app = UnitTestApplication.Start(TestServices.StyledWindow);
+
+        var control = (Control)AvaloniaRuntimeXamlLoader.Load(
+            """
+            <Control
+                xmlns='https://github.com/avaloniaui'
+                xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'
+                xmlns:local='clr-namespace:Avalonia.Markup.Xaml.UnitTests.Xaml;assembly=Avalonia.Markup.Xaml.UnitTests'
+                local:AssignBindingTestControl.AttachedBinding='{Binding SomePath}' />
+            """);
+
+        var binding = AssignBindingTestControl.GetAttachedBinding(control);
+        Assert.NotNull(binding);
+    }
+}
+
+public sealed class AssignBindingTestControl : Control
+{
+    [AssignBinding]
+    public IBinding? ClrBinding { get; set; }
+
+    public static readonly AttachedProperty<IBinding?> AttachedBindingProperty =
+        AvaloniaProperty.RegisterAttached<AssignBindingTestControl, Control, IBinding?>("AttachedBinding");
+
+    public static void SetAttachedBinding(Control obj, [AssignBinding] IBinding? value)
```

Comment:
I think we've traditionally had attributes on setter method rather than parameter. Could you please verify that?

Note that at least Silverlight is [documented](https://learn.microsoft.com/en-us/previous-versions/windows/silverlight/dotnet-windows-silverlight/cc903943(v=vs.95)?redirectedfrom=MSDN) to resolve attached property attributes using its Get method:
>You can attribute a property-level TypeConverter on the Get accessor method. This enables string conversion of the attached property. Applying TypeConverterAttribute to the Get accessor method rather than the Set accessor method may seem nonintuitive, but that is where XAML processors expect to find the type conversion information (if any) for an attached property.

Reply:
You're right, it's the same on WPF.
Example: https://github.com/dotnet/wpf/blob/7df0d6dd0f6d8ec154a3884a37740e93f30c701f/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/RenderOptions.cs#L74

I'll change that.

Reply:
Opened https://github.com/kekekeks/XamlX/pull/127

Reply:
Updated XamlX, changed the attribute and tests to use the getter.